src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/ElasticsearchQueryParser.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.hmcts.ccd.domain.service.search.elasticsearch;
+
+import static java.util.Optional.ofNullable;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import uk.gov.hmcts.ccd.domain.service.common.ObjectMapperService;
+import uk.gov.hmcts.ccd.endpoint.exceptions.ValidationException;
+
+/**
+ * Sample ES query. The parser extracts/sets value for query node.
+ * {
+ * 	"query": {
+ * 		"bool": {
+ * 			"filter": {
+ * 				"match": { "state": "AwaitingPayment"}
+ *                }
+ *            }
+ *       },
+ * 	"sort": {
+ * 		"id": { "order":"asc" }
+ *       }
+ * }
+ */
+class ElasticsearchQueryParser {
+
+    private static final String QUERY_NAME = "query";
+
+    private final ObjectMapperService objectMapperService;
+    private final ObjectNode searchQueryObjectNode;
+
+    ElasticsearchQueryParser(ObjectMapperService objectMapperService, String query) {
+        this.objectMapperService = objectMapperService;
+        this.searchQueryObjectNode = this.objectMapperService.convertStringToObject(query, ObjectNode.class);
+    }
+
+    String extractQueryClause() {
+        return ofNullable(searchQueryObjectNode.get(QUERY_NAME))
+            .map(JsonNode::toString)
+            .orElseThrow(() -> new ValidationException("invalid search query"));
```

Comment:
this is redundant. The search request has already been validated on the controller, so it must have a query entry

Reply:
its harmless though, not a great deal to retain I guess

Reply:
the problem with this, is that it can create confusion. Imagine we add other validations to the search request in the controller. Then we might be tempted to add them here as well, or be undecided. That's how duplication builds up. 
Better to have all the validation of the SearchRequest only in the controller layer and fail fast

Reply:
Moved logic to CaseSearchRequest